src/compiler/transformers/declarations.ts

```diff
@@ -487,7 +487,7 @@ namespace ts {
             | PropertySignature;
 
         function ensureType(node: HasInferredType, type: TypeNode | undefined, ignorePrivate?: boolean): TypeNode | undefined {
-            if (!ignorePrivate && hasModifier(node, ModifierFlags.Private)) {
+            if (!ignorePrivate && hasEffectiveModifier(node, ModifierFlags.Private)) {

```

Comment:
I thought you said earlier that jsdoc wasn't always available during transforms. Did I get that wrong?

Reply:
`parent` isn't always guaranteed during transformations so its unsafe to rely on. In this case, we are generating declarations directly from the parse tree, so `node` *should* have addressable jsdoc and that jsdoc (if it exists) will inform decisions about whether to include syntactic modifiers on the declaration output (which we would want in this case).